src/structlog/threadlocal.py

```diff
@@ -255,6 +258,28 @@ def unbind_threadlocal(*keys: str) -> None:
         context.pop(key, None)
 
 
+@contextlib.contextmanager
+def bound_threadlocal(**kw: Any) -> None:
+    """
+    Bind *kw* to the current thread-local context. Unbind *kw* afterwards &
+    restore overwritten keys. Can be used as a context manager or decorator.
+
+    Use this instead of :func:`~structlog.threadlocal.tmp_bind` when you want
+    the temporary bindings to be applied to the global context (thread-local).
```

Comment:
Let's not point to `tmp_bind`, the new APIs stand for themselves:

```suggestion
    Bind *kw* to the current thread-local context. Unbind or restore *kw*
    afterwards. Do **not** affect other keys.

    Can be used as a context manager or decorator.
```